src/macaron/malware_analyzer/pypi_heuristics/metadata/wheel_absence.py

```diff
@@ -72,7 +75,24 @@ def analyze(self, pypi_package_json: PyPIPackageJsonAsset) -> tuple[HeuristicRes
                 if release_metadata["packagetype"] == self.WHEEL:
                     wheel_present = True
 
-                release_files.append(release_metadata["filename"])
+                name = json_extract(pypi_package_json.package_json, ["info", "name"], str)
+                if name is None:
+                    error_msg = "There is no 'name' field for this package."
+                    logger.debug(error_msg)
+                    raise HeuristicAnalyzerValueError(error_msg)
+
+                # include the pypi inspector link, which uses the same suffix of
+                # packages/{blake2b_256}/file_name
+                inspector_prefix = f"{self.INSPECTOR_PREFIX}{name.lower()}/{version}/"
+                inspector_link = release_metadata["url"].replace(self.PYPI_PREFIX, inspector_prefix)
```

Comment:
If I understand the code correctly, if a project doesn't publish the wheel or tarball, we won't send a HEAD request to check the inspector link, right?

Reply:
Yes, since we iterate over the release data for that version, each item in that is for a separate way of distributing it (`.tar.gz`, `.whl`) so if one of them doesn't exist it won't exist to iterate over, so the HEAD request won't get sent.